src/Microsoft.AspNet.Razor/CodeGenerators/CodeWriter.cs

```diff
@@ -10,32 +9,21 @@ namespace Microsoft.AspNet.Razor.CodeGenerators
     public class CodeWriter : IDisposable
     {
         private static readonly char[] NewLineCharacters = new char[] { '\r', '\n' };
-        private readonly StringWriter _writer = new StringWriter();
-        private bool _newLine;
+
         private string _cache = string.Empty;
-        private bool _dirty = false;
+        private bool _dirty;
 
         private int _absoluteIndex;
         private int _currentLineIndex;
         private int _currentLineCharacterIndex;
 
-        public StringBuilder Builder => _writer.GetStringBuilder();
-
-        public string LastWrite { get; private set; }
+        public StringBuilder Builder { get; } = new StringBuilder();
 
         public int CurrentIndent { get; private set; }
 
-        public string NewLine
-        {
-            get
-            {
-                return _writer.NewLine;
-            }
-            set
-            {
-                _writer.NewLine = value;
-            }
-        }
+        public bool IsAfterNewLine { get; private set; }
+
+        public string NewLine { get; set; } = Environment.NewLine;
```

Comment:
Realize this is output and not input. But are newlines handled correctly when those in the `.cshtml` file don't match `Environment.NewLine` e.g. a Windows-ended file read on a Linux system?